internal/watcher/file/file_watcher_service.go

```diff
@@ -97,8 +97,25 @@ func (fws *FileWatcherService) Watch(ctx context.Context, ch chan<- FileUpdateMe
 	}
 }
 
-func (fws *FileWatcherService) SetEnabled(enabled bool) {
-	fws.enabled.Store(enabled)
+func (fws *FileWatcherService) DisableWatcher(ctx context.Context) {
+	if fws.watcher != nil && fws.watcher.WatchList() != nil {
```

Comment:
Can you add a debug message here to say disabling watcher? Same with the EnableWatcher function.